sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppNavigation.kt

```diff
@@ -15,18 +17,27 @@ fun AppNavigation(
         navController = navController,
         startDestination = startDestination
     ) {
-        composable(AppDestination.Home) {
+        composable(destination = AppDestination.Home) {
             HomeScreen(
-                navigator = { destination -> navController.navigate(destination) }
+                navigator = { destination ->
+                    navController.navigate(destination, destination.parcel)
```

Comment:
Question 🙋‍♀️   @kaungkhantsoe @hoangnguyen92dn,  I found something interesting in this [link](https://gist.github.com/Aidanvii7/46f9f014ba4dc58a2ac10625b198769d#file-navcontrollercomposeutils-kt) It seems like from the example they created a `NavController` extension. Not sure you have tried to follow this. I feel like It will be flexible when we have to add/customize `NavOptions` or `Navigator.Extras` as well, but it might require more refactored. 🙈 

Any insight for this? 🙏  cc @luongvo @Tuubz

Reply:
@Wadeewee I found the approach you mentioned before, and it looks promising with extensions and implementation. But until now, it's out of date with that approach in here:

https://gist.github.com/Aidanvii7/46f9f014ba4dc58a2ac10625b198769d#file-navcontrollercomposeutils-kt-L23
```
    val argumentBundle: Bundle = addedEntry.arguments ?: Bundle().also {
        addedEntry.arguments = it // 👈 This doesn't support var to reassign the argument value
    }
```

And @kaungkhantsoe also mentioned that on the [discussion](https://github.com/nimblehq/android-templates/issues/377#issuecomment-1379951504) 😄 

![image](https://user-images.githubusercontent.com/6950766/215929900-15c9218a-4f02-4246-97b3-f8170f86e24a.png)

Reply:
Ah, I see thanks for your explanation @hoangnguyen92dn 🙇‍♀️